tests/fc_package_argc_argv_int/fc_package_argc_argv_int.c

```diff
@@ -0,0 +1,642 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+#include "macro_utils/macro_utils.h"
+
+#include "testrunnerswitcher.h"
+
+#include "c_logging/xlogging.h"
+
+#include "c_pal/gballoc_hl.h"
+#include "c_pal/gballoc_hl_redirect.h"
+
+#include "com_wrapper/com_wrapper.h"
+
+#include "sf_c_util/common_argc_argv.h"
+
+#include "sf_c_util/fc_package_com.h"
+#include "sf_c_util/fc_package.h"
+
+static bool argc_argv_are_equal(int argc_left, char** argv_left, int argc_right, char** argv_right)
+{
+    if (argc_left != argc_right)
+    {
+        return false;
+    }
+    else
+    {
+        for (int i = 0; i < argc_left; i++)
+        {
+            if (strcmp(argv_left[i], argv_right[i]) != 0)
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+}
+
+static TEST_MUTEX_HANDLE test_serialize_mutex;
+
+TEST_DEFINE_ENUM_TYPE(ARGC_ARGV_DATA_RESULT, ARGC_ARGV_DATA_RESULT_VALUES);
+
+BEGIN_TEST_SUITE(TEST_SUITE_NAME_FROM_CMAKE)
+
+TEST_SUITE_INITIALIZE(suite_init)
+{
+    test_serialize_mutex = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(test_serialize_mutex);
+}
+
+TEST_SUITE_CLEANUP(suite_cleanup)
+{
+    TEST_MUTEX_DESTROY(test_serialize_mutex);
+}
+
+TEST_FUNCTION_INITIALIZE(method_init)
+{
+    if (TEST_MUTEX_ACQUIRE(test_serialize_mutex))
+    {
+        ASSERT_FAIL("Could not acquire test serialization mutex.");
+    }
+}
+
+TEST_FUNCTION_CLEANUP(method_cleanup)
+{
+    TEST_MUTEX_RELEASE(test_serialize_mutex);
+}
+
+/*all tests are actually going against IFabricConfigurationPackage rather than FC_PACKAGE_HANDLE. COM_WRAPPER ROCKS!!!*/
+/*including getting the produced IFabricConfigurationPackage -> argc/argv*/
+
+TEST_FUNCTION(FC_PACKAGE_with_0_sections_succeeds)
+{
+    ///arrange
+    char* argv[] =
+    {
+        CONFIGURATION_PACKAGE_NAME,
+        "CONFIG"
+    };
+    int argc = sizeof(argv) / sizeof(argv[0]);
+    int argc_consumed;
+
+    ///act(1)
+    FC_PACKAGE_HANDLE fc_package = fc_package_create(argc, argv, &argc_consumed);
+
+    ///assert(1)
+    ASSERT_IS_NOT_NULL(fc_package);
+    ASSERT_ARE_EQUAL(int, 2, argc_consumed);
+
+    ///act(2)
+    IFabricConfigurationPackage* obj = COM_WRAPPER_CREATE(FC_PACKAGE_HANDLE, IFabricConfigurationPackage, fc_package, fc_package_destroy);
+
+    ///assert(2)
+    ASSERT_IS_NOT_NULL(obj);
+
```

Comment:
This is breaking our own rules about a test testing one thing. The problem with that is "everybody will want one". As in we'll start doing this more and more and we will end up with long tests where we don't really know what is being tested.

So I'd say let's keep ourselves far away from the temptation :-).

Reply:
Alright, I will break these tests in multiple other tests.